cmd/nitro/init_test.go

```diff
@@ -859,10 +851,351 @@ func TestIsWasmDb(t *testing.T) {
 	for _, testCase := range testCases {
 		name := fmt.Sprintf("%q", testCase.path)
 		t.Run(name, func(t *testing.T) {
-			got := isWasmDB(testCase.path)
+			got := isWasmDb(testCase.path)
 			if testCase.want != got {
 				t.Fatalf("want %v, but got %v", testCase.want, got)
 			}
 		})
 	}
 }
+
+func TestGetConsensusParsedInitMessage(t *testing.T) {
+	t.Parallel()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	stackConfig := testhelpers.CreateStackConfigForTest(t.TempDir())
+	stackConfig.DBEngine = rawdb.DBPebble
+	stack, err := node.New(stackConfig)
+	Require(t, err)
+	defer stack.Close()
+
+	nodeConfig := NodeConfigDefault
+	nodeConfig.Execution.Caching.StateScheme = rawdb.PathScheme
+	nodeConfig.Chain.ID = 42161
+	nodeConfig.Node = *arbnode.ConfigDefaultL2Test()
+	nodeConfig.Init.DevInit = true
+	nodeConfig.Init.DevInitAddress = "0x3f1Eae7D46d88F08fc2F8ed27FCb2AB183EB2d0E"
+	nodeConfig.Init.ValidateGenesisAssertion = false
+
+	l1Client := ethclient.NewClient(stack.Attach())
+
+	chainID := new(big.Int).SetUint64(nodeConfig.Chain.ID)
+
+	// Create dummy chainConfig with chainID
+	chainConfig := params.ChainConfig{}
+	chainConfig.ChainID = chainID
+
+	parsedInitMessage, err := getConsensusParsedInitMsg(ctx, &nodeConfig, chainID, l1Client, chaininfo.RollupAddresses{}, &chainConfig)
+	Require(t, err)
+
+	if parsedInitMessage.ChainId.Uint64() != chainID.Uint64() {
```

Comment:
In this test, chainID provided to getConsensusParsedInitMsg is ignored, the fact that parsedInitMessage.ChainId is equal to chainID is a coincidence.
Here it should compare parsedInitMessage.ChainId with chainConfig.ChainID.